File: utils/date_parser.py
    import re
    import dateparser
    from dateparser.search import search_dates


    def parse_datetime_from_text(text):
        """Извлечь дату и время из текста"""
        result = {
            'date': '',
            'time': '',
            'has_date': False,
            'has_time': False,
        }

        if not text or not text.strip():
            return result

        # Сначала ищем время формата HH:MM через regex 
        time_match = re.search(r'\b(\d{1,2}):(\d{2})\b', text)
        if time_match:
            hour = int(time_match.group(1))
            minute = int(time_match.group(2))
            if 0 <= hour <= 23 and 0 <= minute <= 59:
                result['time'] = f'{hour:02d}:{minute:02d}'
                result['has_time'] = True

        # Ищем дату через dateparser
        try:
            settings = {
                'PREFER_DATES_FROM': 'future',
                'RETURN_AS_TIMEZONE_AWARE': False,
            }
            found = search_dates(text, languages=['ru', 'en'], settings=settings)
            if found:
                _, parsed = found[0]
                result['date'] = parsed.strftime('%Y-%m-%d')
                result['has_date'] = True
                # Если regex не нашёл время, берём из dateparser
                if not result['has_time'] and (parsed.hour or parsed.minute):
                    result['time'] = parsed.strftime('%H:%M')
                    result['has_time'] = True
        except Exception as e:
            print(f'Ошибка парсинга даты: {e}')

        return result

File: utils/date-parser.ts
import * as chrono from 'chrono-node';

export interface ParsedDateTime {
  date: string;
  time: string;
  hasDate: boolean;
  hasTime: boolean;
}

export function parseDateTimeFromText(text: string): ParsedDateTime {
  const result: ParsedDateTime = {
    date: '',
    time: '',
    hasDate: false,
    hasTime: false,
  };

  if (!text || text.trim().length === 0) {
    return result;
  }

  try {
    const parsedResults = chrono.ru.parse(text, new Date(), { forwardDate: true });
    
    if (parsedResults.length > 0) {
      const parsed = parsedResults[0];
      const parsedDate = parsed.start.date();

      // Форматируем дату в YYYY-MM-DD
      const year = parsedDate.getFullYear();
      const month = String(parsedDate.getMonth() + 1).padStart(2, '0');
      const day = String(parsedDate.getDate()).padStart(2, '0');
      result.date = `${year}-${month}-${day}`;
      result.hasDate = true;

      // Форматируем время в HH:MM
      const hours = String(parsedDate.getHours()).padStart(2, '0');
      const minutes = String(parsedDate.getMinutes()).padStart(2, '0');
      result.time = `${hours}:${minutes}`;
      result.hasTime = true;
    }
  } catch (error) {
    console.error('Ошибка парсинга даты:', error);
  }

  return result;
}
